### ISSUE TITLE
Initial commit changes not detected correctly (1.4.0)

### ISSUE THREAD

**Issue by Caeden117** [github-for-unity/Unity#1066]
### Description

I've been previously using Unity Collaborate for a while, and when I found out that more people wanted to contribute to my project (More than what Unity Collaborate allows for free), I picked up GitHub for Unity so that more than 3 people can work on my project.

This is the exact same issue as #1041 (EDIT: and #1044 from the looks of things), however it persists in version 1.4.0 (Even though Release 1.3.1 was said to have fixed this).

### Steps to Reproduce

1. Create an empty, private GitHub repository before installing GitHub for Unity
2. Have a Unity project that doesnt yet have GitHub for Unity configured or installed.
3. Install GitHub for Unity, and input the private GitHub repository as the remote repository in Settings.
4. An initial commit should be created, with the only files being `.gitignore`, `.gitattributes`, and an `Assets` folder with another `.gitignore` inside.
5. Try changing a few files. It does not show up as a changed file in GitHub for Unity's Changes tab.

**Expected behavior:** The initial commit should have everything in the project so far, which allows me to easily push to the remote GitHub repository

**Actual behavior:** Only a few files are commited (`.gitignore`, `.gitattributes`, and the `Assets` folder with another `.gitignore` inside), and changing files do not reflect in GitHub for Unity. I can't push any of my actual Unity files (scripts, graphics, materials, scenes, shaders, etc.) to the GitHub repository.

**Reproduces how often:** Once you have done Steps to Reproduce, it'll stay like that until, what I'd imagine, something else pushes the files to GitHub, or I somehow get rid of everything GitHub for Unity created and restart from scratch (This time publishing a new GitHub repository instead of having one created beforehand).

I've tried pushing my files with GitHub for Visual Studio, however it gives me a `fatal: The remote end hung up unexpectedly`. Closing Visual Studio and re-opening the Unity project does not change any results.

### Additional Information

[I mean, if you really want my log file, it's mainly just me trying to push the darn thing to GitHub because I had a README in the remote repository and Git wasn't liking that.](https://github.com/github-for-unity/Unity/files/3443609/github-unity.log)

**Comment by abhimonk** [comment]
I'm running into this same issue. I've used Github for unity with success in the past, but just tried installing version 1.4.0 in an existing Unity project that did not use github and none of my files are being picked up. When I type 'git status' in the command line, it says that all of the files are untracked. 

I could just git add all the files, but even after doing that they are still not recognized by the "changes" tab in the Github window.

**Comment by Sambardo** [comment]
I'm seeing the same issue. Did you find a convenient fix to get them to be recognized in the add on?

**Comment by Caeden117** [comment]
Sadly not. What I did was completely remove everything GitHub for Unity generated (Hidden `.git` folder, files, etc.) and make a new local repository using Visual Studio's GitHub extension, which seemed to work flawlessly, minus the [`.gitignore` super helpful for Unity projects](https://github.com/github/gitignore/blob/master/Unity.gitignore).

I soon removed the Unity extension as I was already used to managing my local repo through the Visual Studio extension.

I thought this was an issue with my Unity version, and that GitHub for Unity lost compatibility for Unity 5. Last month I finally made the jump to 2018.3 and would take another look at this extension soon, however from the looks of things you guys are experiencing the same problems on similar or newer Unity versions.

**Comment by jwvanderbeck** [comment]
Same problem.

The really confuding aspect is GitHub itself seems to be adding the .gitignore file to the Assets folder which of course tells git to ignore the entire project essentially.

At first I thought this was from Collab but I deleted the file, deleted Git, reset everything then reinstalled Git and reinitialized the repo and GitHuib for Unity re-added the gitignore to the Asset folder.

So Um.. why is GitHub for Unity trying to essentially ignore the whole project?

EDIT: After researching this some more I was wrong.  The gitignore file it creates in he assets folder is empty so I don't think that does anything.  Still not sure why it creates one there and still not sure why Github in Unity doesn't see any of the files, but it isn't for the reason I thought it was.

If I set everything up using Github Desktop first then Github Unity works fine, but setting up through Github Unity does not.
